pmd-core/src/main/java/net/sourceforge/pmd/lang/AbstractParser.java

```diff
@@ -34,4 +38,13 @@ public TokenManager getTokenManager(String fileName, Reader source) {
     }
 
     protected abstract TokenManager createTokenManager(Reader source);
+
+    @Deprecated
+    public static Node doParse(Parser parser, String fileName, Reader source) {
```

Comment:
I'll leave this static method in PMD 6 as is... although it looks weird...

Reply:
I don't think we need an Abstract Parser or a default implementation in Parser in pmd 7, since the AstInfo already preserves the file name. In pmd 7, probably nothing needs to change about the call sites of Parser#parse, just the implementation of the xpath function

Reply:
Ok, I'll see once I'll merge master into pmd/7.0.x....